Display Recode

DalwynWasTaken opened the pull request:
Breaking changes to Display section of ChatTriggers. I'm merging it here so it can be added later. Feel free to polish it if you feel like something is missing/wrong

DalwynWasTaken committed:
Merge remote-tracking branch 'Squagward/breaking' into breaking

DalwynWasTaken commented:
Not sure how to remove the Merge commit, but it shouldn't effect a thing. And changelog for Display is in the Display Recode commit

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -1,44 +1,92 @@
     private var backgroundColor: Long = 0x50000000
-    private var textColor: Long = 0xffffffff
-    private var background = DisplayHandler.Background.NONE
+    private var textColor: Long = 0xFFFFFFFF
```
Revert these back to 0xffffffff to be consistent with the other files

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -77,32 +125,46 @@ abstract class Display() {
         this.lines = CopyOnWriteArrayList(lines)
     }
 
-    @JvmOverloads
```
@JvmOverloads needs to stay since the index is optional.

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -115,6 +177,20 @@ abstract class Display() {
+     *
+     * @param line The index of the line or the DisplayLine to remove
+     */
+    fun removeLine(line: Any) = apply {
```
Add a string option here

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -125,111 +201,79 @@ abstract class Display() {
 
     fun show() = apply {
-        setShouldRender(true)
+        this.visible = true
```
Remove all unneeded uses of `this`, I might have missed a couple.

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -125,111 +201,79 @@ abstract class Display() {
                 "}"
-
-}
+}
```
Add a new line at the end and remove the empty line above.

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/DisplayHandler.kt:
```diff
@@ -55,4 +56,5 @@ object DisplayHandler {
     }
-}
+
+}
```
Same thing here, remove the new line on line 59 and add one after line 60

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/DisplayLine.kt:
```diff
@@ -54,18 +68,13 @@ abstract class DisplayLine {
-        MouseListener.registerDraggedListener { deltaX, deltaY, x, y, button ->
-            if (shouldRender)
-                onDragged?.trigger(arrayOf(deltaX, deltaY, x, y, button))
-        }
```
Why was dragged removed?

DalwynWasTaken replied:
It has absolutely no use other than adding a check to the dragged method. Just allow the user to define their own dragged register at this point.

camnwalter replied:
Ok yeah that makes sense.  Register dragged for a single DisplayLine doesn't make too much sense.

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/DisplayLine.kt:
```diff
@@ -204,13 +234,14 @@ abstract class DisplayLine {
+                "text=$text, textColor=$textColor, align=$align" +
+                "backgroundType=$backgroundType, backgroundColor=$backgroundColor" +
                 "}"
 }
```
Add a new line at the very end

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -77,32 +125,46 @@ abstract class Display() {
+        }
 
-        lines[index] = when (line) {
+        this.lines[index] = when (line) {
```
Remove unneeded `this`

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -125,111 +201,79 @@ abstract class Display() {
-    fun setRenderLoc(renderX: Float, renderY: Float) = apply {
-        this.renderX = renderX
-        this.renderY = renderY
+        this.lines.clear()
```
Remove unneeded `this`

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -125,111 +201,79 @@ abstract class Display() {
 
     internal abstract fun createDisplayLine(text: String): DisplayLine
 
+    internal abstract fun getLoader(): ILoader
```
Inside `Impl.kt`, add `override fun getLoader` for JSDisplay like the other classes in there.

Squagward commented:
> Not sure how to remove the Merge commit, but it shouldn't effect a thing. And changelog for Display is in the Display Recode commit

I can fix the merge commit whenever I merge this in, don't worry about that.

DalwynWasTaken commented:
I don't understand the new line stuff so I based it off of what the current CT main branch looks like. And I finally understand why some properties have this and others don't. I didn't realize that until I went through to remove unnecessary this's

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -77,15 +125,38 @@ abstract class Display() {
+    fun getLine(line: Any): Any? {
+        return when (line) {
+            is Int -> this.lines[line]
+            is String -> this.lines.find { it.getText() == line }
```
Remove `this`

camnwalter commented on src/main/kotlin/com/chattriggers/ctjs/minecraft/objects/display/Display.kt:
```diff
@@ -115,6 +184,21 @@ abstract class Display() {
+        when (line) {
+            is Int -> lines.removeAt(line)
+            is DisplayLine -> lines.remove(line)
+            is String -> lines.forEach { if (it.getText() == line) lines.remove(it) }
```
```suggestion
            is String -> lines.removeIf { it.getText() == line }
```

camnwalter replied:
Actually, that won't do it since `getText` returns a `Text`, not a string, but still.

DalwynWasTaken committed:
Display Recode

I deleted all the code and went through it step by step to see what was needed and what wasn't. I didn't change the base drawing logic.

Changes:
+ Added documentation for the config object and where needed
+ Added postDraw and preDraw triggers so you can draw above and under the display in relativity to the display
+ Added remove line (index, string or DisplayLine)
+ Added getIndexOfLine method

= setShouldRender -> show/hide
= renamed Background to BackgroundType
= renamed renderX/Y to X/Y

- removed registerDragged and all code associated

Fixes and patches (rebased)

- Added getLoader override to Impl.kt
- Added back JvmOverloads
- Made removeLine work with string
- Made getLine work with string

Tested PostDraw and PreDraw registers. They work as intended for as far as my tests go

Squagward commented:
LGTM thank you 👍
